src/exchange.c

```diff
@@ -291,7 +292,7 @@ static int p11prov_ecdh_derive(void *ctx, unsigned char *secret,
     }
 
     P11PROV_debug("ECDH derived hey handle: %lu", secret_handle);
-    FA_ASSIGN_ALL(attrs[0], CKA_VALUE, &secret, &secret_len, false, true);
+    FA_SET_BUF_VAL(attrs, num, CKA_VALUE, secret, secret_len, false, true);
     ret = p11prov_fetch_attributes(ecdhctx->provctx, session, secret_handle,
                                    attrs, 1);
```

Comment:
```suggestion
                                   attrs, num);
```

Reply:
I would rather have larger diff than magic numbers around that you need to change when you add more attrs. But it might be just my preference.